mpt_trie/src/partial_trie.rs

```diff
@@ -434,7 +437,8 @@ where
     V: Into<ValOrHash>,
 {
     let mut root = N::new(Node::Empty);
-    root.extend(nodes);
-
-    root
+    match root.extend(nodes) {
+        Ok(_) => root,
+        Err(e) => panic!("Error extending trie: {}", e),
```

Comment:
Could we not have an actual error here rather than just a panic? Return `Result<>` and have an error variant for "can't extend the tree for whatever reason"